src/api.rs

```diff
@@ -234,6 +234,21 @@ impl ThreadPool {
             job_a.into_result()
         }
     }
+
+    /// Returns the number of threads in the thread pool.
+    pub fn num_threads(&self) -> usize {
+        self.registry.num_threads()
+    }
+
+    /// Returns the index of the current thread in the thread pool. Panics if
+    /// the current thread is not in this thread pool.
```

Comment:
This actually panics if the current thread is not a Rayon thread, but doesn't check that the current thread is in *this* thread-pool. You would have to compare the registries; I forget if the `RegistryId` code which would enable this is in master, I think it is only in https://github.com/nikomatsakis/rayon/pull/141